easybuild/tools/docs.py

```diff
@@ -160,3 +147,134 @@ def avail_easyconfig_params(easyblock, output_format):
         FORMAT_TXT: avail_easyconfig_params_txt,
     }
     return avail_easyconfig_params_functions[output_format](title, grouped_params)
+
+def generic_easyblocks(path_to_examples, common_params={}, doc_functions=[]):
+    """
+    Compose overview of all generic easyblocks
+    """
+    modules = import_available_modules('easybuild.easyblocks.generic')
+    docs = []
+    all_blocks = []
+
+    # get all blocks
+    for m in modules:
+        for name,obj in inspect.getmembers(m, inspect.isclass):
+            eb_class = getattr(m, name)
+            # skip imported classes that are not easyblocks
+            if eb_class.__module__.startswith('easybuild.easyblocks.generic') and eb_class not in all_blocks:
+                all_blocks.append(eb_class)
+
+    for eb_class in all_blocks:
+        docs.append(doc_easyblock(eb_class, path_to_examples, common_params, doc_functions, all_blocks))
+
+    toc = ['.. contents:: Available generic easyblocks', '    :depth: 1', '']
```

Comment:
drop the `Available generic easyblocks`, we should have a page title in place instead (above the `.. contents::`)